Google.PowerShell/Sql/GcSqlInstanceCmdlets.cs

```diff
@@ -987,4 +988,387 @@ protected override void ProcessRecord()
+    /// </para>
+    /// <para>
+    /// Caution: If "Update" is true, this is not a partial update, so you must include values for all the settings that you want to retain.
+    /// </para>
+    /// <example>
+    ///   <para>
+    ///   Patches the SQL Instance "myInstance" (with tier "db-n1-standard-1" and setting version of 15)
+    ///   so that it can have maintenance on Monday at 22:00.
+    ///   </para>
+    ///   <para><code>
+    ///     PS C:\> Update-GcSqlInstance "myInstance" "db-n1-standard-1"`
+    ///         15 -MaintenanceWindowDay 1 -MaintenanceWindowHour "22:00" -Project "testing" 
+    ///   </code></para>
+    ///   <br></br>
+    ///   <para>(If successful, the command returns the resource for the updated instance.)</para>
+    /// </example>
+    /// <example>
+    ///   <para>
+    ///   Updates the SQL Instance "myInstance" (with tier "db-n1-standard-1" and setting version of 18)
+    ///   so that its settings default.
+    ///   </para>
+    ///   <para><code>
+    ///     PS C:\> Update-GcSqlInstance "myInstance" "db-n1-standard-1" 18 -Update
+    ///   </code></para>
+    ///   <br></br>
+    ///   <para>(If successful, the command returns the resource for the updated instance.)</para>
+    /// </example>
+    /// </summary>
+    [Cmdlet(VerbsData.Update, "GcSqlInstance", DefaultParameterSetName = ParameterSetNames.ByName)]
+    public class UpdateGcSqlInstanceCmdlet : GcSqlCmdlet
+    {
+        private class ParameterSetNames
+        {
+            public const string ByName = "ByName";
+            public const string ByInstance = "ByInstance";
+        }
+
+        /// <summary>
+        /// <para type="description">
+        /// Name of the project. Defaults to the Cloud SDK configuration for properties if not specified.
+        /// </para>
+        /// </summary>
+        [Parameter(ParameterSetName = ParameterSetNames.ByName)]
+        [ConfigPropertyName(CloudSdkSettings.CommonProperties.Project)]
+        public string Project { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The name of the instance to be updated/patched.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = true, Position = 0,
+            ParameterSetName = ParameterSetNames.ByName)]
+        public string Instance { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The tier of service for this instance, for example "db-n1-standard-1".
+        /// Pricing information is available at https://cloud.google.com/sql/pricing.
+        /// Get-GcSqlTiers will also tell you what tiers are available for your project.
+        /// </para>
+        /// </summary>
+        [Parameter(Position = 1, Mandatory = true, ParameterSetName = ParameterSetNames.ByName)]
+        [Parameter(Position = 0, Mandatory = true, ParameterSetName = ParameterSetNames.ByInstance)]
+        public string Tier { get; set; }
```

Comment:
If it's not specified the operation fails, but I suppose I could pull it from the instance.